### PR TITLE
Support for strong pokemon

### PR DESCRIPTION
**Pull request by Mygod**

### CODE REVIEW

File: decoder/pokemon.go [hunk 1]
```diff
@@ -724,8 +724,10 @@ func (pokemon *Pokemon) addEncounterPokemon(ctx context.Context, db db.DbDetails
 	var level int64
 	if cpMultiplier < 0.734 {
 		level = int64(math.Round((58.35178527*cpMultiplier-2.838007664)*cpMultiplier + 0.8539209906))
-	} else {
+	} else if cpMultiplier < .795 {
 		level = int64(math.Round(171.0112688*cpMultiplier - 95.20425243))
+	} else {
+		level = int64(math.Round(199.99995231630976*cpMultiplier - 118.05996066890287))
```
Comment (Fabio1988) [inline decoder/pokemon.go:730]: I am just curious who is doing those calculations? cpMultiplier is listed in Gamemaster, but where do the other values come from? :)

Comment (Mygod) [reply decoder/pokemon.go:730]: Asked chatgpt to write an interpolation. I can't be bothered to read what interpolation it is but I can send you the source code. (I also wrote tests to make sure it works.)

File: decoder/pokemon.go [hunk 2]
```diff
@@ -724,8 +724,10 @@ func (pokemon *Pokemon) addEncounterPokemon(ctx context.Context, db db.DbDetails
 	var level int64
 	if cpMultiplier < 0.734 {
```
Comment (Fabio1988) [inline decoder/pokemon.go:725]: I wonder why we ever used those "highly complex" multiplications, I think this was already a thing in RDM.

We know exactly the cp-multiplier to level mapping from GM, right? should we somehow hard-code them? Gohbem already knows the gamemaster for doing the pvp calculations, should we add a method to just return level on cp_multiplier input? @jfberry any thoughts?

Comment (Fabio1988) [reply decoder/pokemon.go:725]: or we try to hard-code the level lookup?

```golang
type LevelMultiplier struct {
	Level      int
	Multiplier float64
}

table := []LevelMultiplier{
		{1, 0.094}, {2, 0.16639787}, {3, 0.21573247}, {4, 0.255720049},
		{5, 0.290249884}, {6, 0.321087599}, {7, 0.349212676}, {8, 0.375235587},
		{9, 0.399567276}, {10, 0.4225}, {11, 0.443107545}, {12, 0.462798387},
		{13, 0.481684953}, {14, 0.499858439}, {15, 0.517393947}, {16, 0.534354329},
		{17, 0.550792694}, {18, 0.56675452}, {19, 0.582278907}, {20, 0.5974},
		{21, 0.612157285}, {22, 0.626567125}, {23, 0.640652955}, {24, 0.654435635},
		{25, 0.667934}, {26, 0.68116492}, {27, 0.694143653}, {28, 0.706884205},
		{29, 0.719399095}, {30, 0.7317}, {31, 0.737769485}, {32, 0.743789434},
		{33, 0.749761045}, {34, 0.755685508}, {35, 0.761563838}, {36, 0.767397165},
		{37, 0.773186505}, {38, 0.77893275}, {39, 0.784637}, {40, 0.7903},
		{41, 0.7953}, {42, 0.8003}, {43, 0.8053}, {44, 0.8103}, {45, 0.8153},
		{46, 0.8203}, {47, 0.8253}, {48, 0.8303}, {49, 0.8353}, {50, 0.8403},
		{51, 0.8453}, {52, 0.8503}, {53, 0.8553}, {54, 0.8603}, {55, 0.8653},
	}
```

**Review by Ppeepost4489 — CHANGES_REQUESTED** [review]
651eec24714983bf71bdc0fe94beafc607cc244d

**Pull request merged**